anon/elgam.go

```diff
@@ -0,0 +1,82 @@
+package anon
+
+import (
+	"bytes"
+	"errors"
+	"crypto/cipher"
+	"github.com/dedis/crypto/abstract"
+)
+
+type basicSig struct {
```

Comment:
You don't really need this typedef, since you're going to be encoding the signature into a []byte before it leaves the Sign function anyway.